### DISCUSSION TITLE
error "Github ssl key The private key is not a valid key or has a passphrase." when enabling TLS in Enterprise server

### DISCUSSION THREAD

**Discussion by mm808** [community/community#52192] [Enterprise]
Hi all,
I have created a new VM running GitHub 3.8 Enterpise. As part of the set up I want to enable TLS. However when I upload the certificate and private key our IT team gave me, I get the error "Github ssl key The private key is not a valid key or has a passphrase." I am certain the key does not have a password and is not encrypted as I viewed it in a txt editor. The certificate is a wildcard (*.company.com) cert if that makes a difference.
Any ideas why I get this error and how to get around it?

**Comment by SMelidoni** [comment]
Hi there @mm808 👋🏽

It seems like you might be facing an issue with the format or content of the private key file. To resolve this issue, please follow the steps below:

1. **Verify the private key format:**
GitHub Enterprise supports private keys in the PEM format. Make sure your private key is in PEM format, which should look like this:

 ```
    -----BEGIN PRIVATE KEY-----
    (Your Private Key content here)
    -----END PRIVATE KEY-----
 ```

If your private key is in another format, such as PKCS#12, you may need to convert it to PEM format using a tool like OpenSSL.

2. **Verify the private key and certificate match:**
It's important to ensure that the provided private key matches the certificate. You can use the following OpenSSL commands to check if they match:

For the private key:
`openssl rsa -in private.key -noout -modulus | openssl md5`

For the certificate:
`openssl x509 -in certificate.crt -noout -modulus | openssl md5`

Both commands should output the same hash if they match.

3. **Check for additional issues in the private key:**
If the private key format is correct, and it matches the certificate, there might be an issue with the content of the private key file. You can use OpenSSL to check the private key as follows:

`openssl rsa -in private.key -check`

If there's an issue with the private key, you may need to request a new one from your IT team.

4. **Double-check the certificate:**
Make sure the certificate you've received is the correct one and is in the correct format (PEM). It should look like this:

 ```
    -----BEGIN CERTIFICATE-----
    (Your Certificate content here)
    -----END CERTIFICATE-----
 ```

If you've checked all of the above and still encounter the issue, you might want to reach out to your IT team to verify the certificate and private key provided to you or check the GitHub Enterprise documentation for further information.

Good luck! Let me know if you need more assistance.
